src/main/java/ikura/Database.java

```diff
@@ -0,0 +1,242 @@
+// Database.java
+// Copyright (c) 2020, zhiayang, Apache License 2.0.
+
+package ikura;
+
+import java.util.List;
+import java.util.Optional;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+import java.util.function.Function;
+import java.time.format.DateTimeFormatter;
+
+import ikura.task.Task;
+import ikura.task.Todo;
+import ikura.task.Event;
+import ikura.task.Deadline;
+import ikura.task.DatedTask;
+
+import ikura.util.Either;
+import ikura.util.InvalidInputException;
+import ikura.util.InvalidDatabaseException;
+
+
+/**
+ * A class encapsulating the serialisation and deserialisation of the list of tasks.
+ */
+public class Database {
+
+    private static final int DATABASE_VERSION = 1;
+
+    private final String databasePath;
+
+    /**
+     * Constructs a new database, using the given path as the backing file. The file does
+     * not have to exist, but it should be writeable.
+     *
+     * @param path the path to the file.
+     */
+    public Database(String path) {
+        this.databasePath = path;
+    }
+
+    /**
+     * Load the list of tasks from the database.
```

Comment:
Perhaps you could use Loads instead of Load to be consistent with the coding standard?

Reply:
are you referring to the javadoc? in that case ya good point